src/PubSub/Queue.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+namespace Myli\PlainJobs\PubSub;
+
+use Illuminate\Queue\Queue as IlluminateQueue;
+use Google\Cloud\PubSub\Topic;
+use Google\Cloud\PubSub\Message;
+use Google\Cloud\PubSub\PubSubClient;
+use Illuminate\Contracts\Queue\Queue as QueueContract;
+use Myli\PlainJobs\Jobs\DispatcherJob;
+use Illuminate\Support\Facades\Config;
+
+class Queue extends IlluminateQueue implements QueueContract
+{
+    /**
+     * The PubSubClient instance.
+     *
+     * @var \Google\Cloud\PubSub\PubSubClient
+     */
+    protected $pubsub;
+
+    /**
+     * Default queue name.
+     *
+     * @var string
+     */
+    protected $default;
+
+    /**
+     * Create a new GCP PubSub instance.
+     *
+     * @param \Google\Cloud\PubSub\PubSubClient $pubsub
+     * @param string                            $default
+     */
+    public function __construct(PubSubClient $pubsub, string $default = null)
+    {
+        $this->pubsub  = $pubsub;
+        $this->default = $default;
+    }
+
+    /**
+     * Get the size of the queue.
+     * PubSubClient have no method to retrieve the size of the queue.
+     * To be updated if the API allow to get that data.
+     *
+     * @param  string $queue
+     * @return int
+     */
+    public function size($queue = null)
+    {
```

Comment:
Huh????

Reply:
We must implement the interface Illuminate\Contracts\Queue\Queue, and we can't know how much messages are left in a PubSub queue